text/0000-scriptless-buildpacks.md

```diff
@@ -0,0 +1,211 @@
+run = [ "<string>" ]
+
+[[buildpack.build.layers]]
+id = "<string>"
+cache = "<boolean>"
+launch = "<boolean>"
+build = "<boolean>"
+run = [ "<string>" ]
+
+  [buildpack.build.layers.metadata]
+  key = "<string>" # arbitrary elements that will be put directly into the layer metadata
+
+  [buildpack.build.layers.env]
+  KEY = "<string>" # env vars that will be put into `<layer_dir>/env`
+
+  [[buildpack.build.layers.profile]]
+  # each table in this array will be turned into a <layer_dir>/profile.d script
+  script = "<string>"
+
+[[buildpack.build.launch.processes]]
+# each table in this array will be turned into an entry in `launch.toml`
+type = "<string>"
+command = "<string>"
+```
+
+The lifecycle will use the `layer.id` value to determine if the layer should be regenerated. If the layer is already present (because of caching) and the metadata matches it will skip all other behavior defined in this table. This behavior is effectively described by the following Bash code:
+
+```bash
+if [[ -f ${layers_dir}/${layer_id}.toml ]]; then
+  if [[ "$(cat ${layers_dir}/${layer_id}.toml)" != "${derived_layer_toml}" ]]; then
+   # execute `run` and other directives
+  fi
+fi
+```
+
+Thus, if you want to force a layer to regenerate, the `[buildpack.build.layers.metadata]` must contain appropriate keys to determine if they layer has changed.
+
+The commands defined in any `run` element will have access to script arguments (i.e. `$1`, `$2`, etc) as defined in the [Buildpack Spec](https://github.com/buildpack/spec/blob/master/buildpack.md), as well as other environment variables the lifecycle makes available.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The following examples demonstrate about a scriptless buildpack works in practice (from the buildpack author persona).
+
+## Example: JRuby Warbler buildpack
+
+This buildpack is used to run a `rake` command after a JRuby build:
+
+```toml
+[buildpack]
+id = "io.buildpacks.warbler"
+name = "Warbler Buildpack"
+version = "0.0.9"
+
+[buildpack.detect]
+requires = [
+  "jruby",
+  "warbler",
+  "bundler"
+]
+provides = [
+  "war"
+]
+
+# if there is a bin/build, this will always run first.
+[buildpack.build]
+run = [
+  "rake war",
+  "rake db:migrate"
+]
+
+[[buildpack.build.launch.processes]]
+type = "web"
+command = "java -jar myapp.war"
+```
+
+The buildpack requires `jruby`, `warbler`, and `bundler` be installed by previous buildpacks. It then provides a `war` file.
+
+The build will run two `rake` commands, and then it generates a default `web` command. But no layers are created.
+
+## Example: JDK buildpack
+
+This buildpack is functionally similar to the [sample Java buildpack](https://github.com/buildpack/samples/blob/master/java-buildpack/).
+
+```toml
+
+[buildpack]
+id = "io.buildpacks.zulu"
+name = "Azul Zulu Buildpack"
+version = "0.0.9"
+
+[buildpack.detect]
+provides = [
+  "jdk"
+]
+
+[[buildpack.build.layers]]
+id = "jdk"
+cache = true
+launch = true
+build = true
+run = [
```

Comment:
How do you write metadata from here? How do you read the build plan?

Reply:
i would support positional args like `$1` below this